### PR TITLE
Installer pod fixes

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes this case: https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-4.0/4546

Also switch the installer client to protobuf when fetching secrets/configs.

### CODE REVIEW

File: pkg/operator/staticpod/installerpod/cmd.go [hunk 1]
```diff
@@ -302,8 +302,17 @@ func (o *InstallOptions) copyContent(ctx context.Context) error {
...
 
 func (o *InstallOptions) Run(ctx context.Context) error {
-	eventTarget, err := events.GetControllerReferenceForCurrentPod(o.KubeClient, o.Namespace, nil)
-	if err != nil {
+	var eventTarget *corev1.ObjectReference
+
+	// Poll for the pod here to prevent flakes when the API server is temporary down or connection refused errors.
+	if err := utilwait.PollImmediateUntil(200*time.Millisecond, func() (bool, error) {
+		var getReferenceErr error
+		eventTarget, getReferenceErr = events.GetControllerReferenceForCurrentPod(o.KubeClient, o.Namespace, nil)
```
Comment (sttts) [inline pkg/operator/staticpod/installerpod/cmd.go]: why not err

File: pkg/operator/staticpod/installerpod/cmd.go [hunk 2]
```diff
@@ -302,8 +302,17 @@ func (o *InstallOptions) copyContent(ctx context.Context) error {
...
+		if getReferenceErr != nil {
```
Comment (sttts) [inline pkg/operator/staticpod/installerpod/cmd.go]: are there any errors that we want to pass through? NotFound maybe.

File: pkg/operator/staticpod/installerpod/cmd.go (hunk 1 above)
Comment (mfojtik) [reply pkg/operator/staticpod/installerpod/cmd.go]: don't like shadowing

File: pkg/operator/staticpod/installerpod/cmd.go (hunk 2 above)
Comment (mfojtik) [reply pkg/operator/staticpod/installerpod/cmd.go]: sure

**Pull request merged**